src/__init__.py

```diff
@@ -45,54 +45,40 @@ def compute(ampa: np.ndarray, t_ampa_ms, gaba: np.ndarray, t_gaba_ms, t_eval_ms:
         lfp = wslfp.compute(..., t_gaba_ms=[multiple, gaba, measurements], t_eval_ms=[a, whole, bunch, of, timepoints])
         """
 
-        try:
-            _check_timepoints(t_ampa_ms, t_gaba_ms, t_eval_ms)
-            
-
-
-    def compute_ampa_time(t_ampa_ms, tau_ampa):
-        ampa_time_arr = np.array(t_ampa_ms)
-        for x in ampa_time_arr.size - 1:
-            ampa_time_arr[x] = t_ampa_ms[x] - tau_ampa
-        return ampa_time_arr
-
-# use numpy array instead of loop
-
-    def compute_gaba_time(t_gaba_ms, tau_gaba):
-        gaba_time_arr = np.array(t_gaba_ms)
-        for x in gaba_time_arr.size - 1:
-            gaba_time_arr[x] = t_gaba_ms[x] - tau_gaba
-        return gaba_time_arr
-
-    def compute_ampa(ampa:np.ndarray, t_ampa_ms, tau_ampa):
-        ampa_arr = np.array(ampa)
-        time = compute_ampa_time(t_ampa_ms, tau_ampa)
-        for x in time.size:
-            ampa_arr[x] = ampa[x] * time[x]
-        return ampa_arr
-
-    def compute_gaba(gaba:np.ndarray, t_gaba_ms, tau_gaba):
-    gaba_arr = np.array(gaba)
-    time = compute_gaba_time(t_gaba_ms, tau_gaba)
-    for x in time.size: #no for loop
-        gaba_arr[x] = gaba[x] * time[x] #should be gaba at that time
-    return gaba_arr
-
-    def sum_ampa(ampa, t_ampa_ms, tau_ampa):
-        ampa_sum = 0
-        ampa_arr = compute_ampa(ampa, t_ampa_ms, tau_ampa)
-        for ampa_curr in ampa_arr:
-            ampa_sum += ampa_curr
-        return ampa_sum
-
-    def sum_gaba(gaba, t_gaba_ms, tau_gaba):
-        gaba_sum = 0
-        gaba_arr = compute_gaba(gaba, t_gaba_ms, tau_gaba)
-        for gaba_curr in gaba_arr:
-            gaba_sum += gaba_curr
-        return gaba_sum
-
-    def _check_timepoints(t_ampa_ms, t_gaba_ms, tau_ampa, tau_gaba, t_eval_ms, I_ampa):
+        #try:
+           # _check_timepoints(t_ampa_ms, t_gaba_ms, t_eval_ms)
+    
+    # 1. combine check timepoints to compute curr
+    # 2.
+    
+    #substract tau from t_eval NEW CODE
+    def _compute_gaba_curr(self, gaba, t_gaba_ms, tau_gaba, t_eval_ms): # evaluate at 1 timepoint, so t_eval_ms is a float
+        gaba = np.array(gaba)
+        #row = np.where(t_gaba_ms == t_eval_ms) 
+        # #use interpolation
+        gaba_interp = self._check_gaba_timepoints(gaba, t_gaba_ms, t_eval_ms, tau_gaba)
+        
+        return gaba_interp(t_eval_ms) #return gaba currents of all neurons at t_eval
+
+    #def compute_ampa_curr(self, ampa, t)
+    
+
+    def _compute_ampa_curr(self, ampa, t_ampa_ms, tau_ampa, t_eval_ms): 
```

Comment:
why have separate _compute and _check, when _compute does little more than call _check?